app/routes.py

```diff
@@ -1,2 +1,277 @@
+
+        return jsonify(task_response)
+    elif request.method == 'POST':
+        request_body = request.get_json()
+        if "title" not in request_body:
+            return {
+                "details": "Invalid data"
+            }, 400
+        elif "description" not in request_body:
+            return {
+                "details": "Invalid data"
+            }, 400
+        elif "completed_at" not in request_body:
+            return {
+                "details": "Invalid data"
+            }, 400
+        stringify_format = "%a, %d %b %Y %H:%M:%S %Z"
+        if request_body["completed_at"]:
+
+            new_task = Task(
+                title = request_body['title'],
+                description = request_body['description'], 
+                completed_at = datetime.strptime(request_body["completed_at"], stringify_format)
+            )
+        else:
+             new_task = Task(
+                title = request_body['title'],
+                description = request_body['description'], 
+            )
+        db.session.add(new_task)
+        db.session.commit()
+        return{
+            'task':{
+                'id': new_task.task_id,
+                'title': new_task.title,
+                'description': new_task.description,
+                'is_complete': new_task.completed_at != None
+            }      
+        },201
+
+
+@task_list_bp.route('/<task_id>', methods=['GET','PUT', 'DELETE'])
+def handle_task(task_id):  # same name as parameter route
+    task = Task.query.get(task_id)
+    if not task:
+        return "", 404
+
+    if request.method == 'GET':
+        return({
+            'task': task.serialize()
+        })
+    elif request.method == 'DELETE':
+        db.session.delete(task)
+        db.session.commit()
+        return({
+            "details": f'Task {task_id} "{task.title}" successfully deleted' 
+        },200)
+        
+
+    elif request.method =='PUT':
+        request_body = request.get_json()
+        if 'title' in request_body:
+            task.title = request_body['title']
+        if 'description' in request_body:
+            task.description = request_body['description']
+        if 'completed_at' in request_body:
+            task.completed = request_body['completed_at']
+        if 'goal_id' in request_body:
+            task.goal_id= request_body['goal_id']
+        db.session.commit()
+        return({
+            'task': task.serialize()
+        },200)
+
+@task_list_bp.route('/<task_id>/mark_complete',methods = ['PATCH'])
+def mark_complete_task(task_id):
+    task = Task.query.get(task_id)
+    if not task:
+        return "", 404
+    task.completed_at = datetime.utcnow()
+    task.notify_slack()
+    return{
+        'task':{
+            'id': task.task_id,
+            'title': task.title,
+            'description': task.description,
+            'is_complete': task.completed_at != None
+        }      
```

Comment:
Notice that this code is repeated below. This is a great place to use an instance method.